src/widgets/menu.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2013-2015, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::ptr;
+
+use cast::{GTK_MENU, GTK_MENU_ITEM, GTK_MENU_SHELL};
+use gdk;
+use glib::translate::*;
+use traits::FFIWidget;
+use ffi;
+
+struct_Widget!(Menu);
+impl_TraitWidget!(Menu);
+
+impl Menu {
+    pub fn new() -> Option<Menu> {
+        let tmp_pointer = unsafe { ffi::gtk_menu_new() };
+        check_pointer!(tmp_pointer, Menu)
+    }
+
+    //#[cfg(gtk_3_4)]
+    //pub fn new_from_model<T: Upcast<gio::MenuModel>>(model: &T) -> Menu {
+    //    unsafe { TODO: call ffi:gtk_menu_new_from_model() }
+    //}
+
+    // Should be implemented in a MenuShell trait eventually
```

Comment:
Wait, doesn't `MenuShellTrait`, which is `impl`'d below already have this method?